pom.xml

```diff
@@ -1863,5 +1863,56 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
+            <id>disable-check-spi-dependencies</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>io.trino</groupId>
+                        <artifactId>trino-maven-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>default-check-spi-dependencies</id>
+                                <phase>none</phase>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <profile>
+            <id>gib</id>
+            <properties>
+                <!-- the *local* master, not refs/remotes/... -->
+                <gib.referenceBranch>master</gib.referenceBranch>
+            </properties>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>com.vackosar.gitflowincrementalbuilder</groupId>
+                        <artifactId>gitflow-incremental-builder</artifactId>
+                        <version>3.15.0</version>
+                        <extensions>true</extensions>
+                        <configuration>
+                            <disableIfBranchMatches>master</disableIfBranchMatches>
+                            <compareToMergeBase>true</compareToMergeBase>
+                            <uncommitted>true</uncommitted>
+                            <untracked>true</untracked>
+                            <excludePathsMatching>README.md</excludePathsMatching>
+                            <buildDownstream>true</buildDownstream>
+                            <buildUpstream>true</buildUpstream>
+                            <buildUpstreamMode>impacted</buildUpstreamMode>
+                            <!-- skip tests and checks for upstream modules since they have not been modified but are still required to be built -->
+                            <skipTestsForUpstreamModules>true</skipTestsForUpstreamModules>
+                            <argsForUpstreamModules>-Dmaven.source.skip=true -Dair.check.skip-all</argsForUpstreamModules>
```

Comment:
skip javadocs too?

Reply:
Also `gib.skipIfPathMatches` might be helpful to ensure that change in some files (e.g. `.github/workflows`?) causes gib to consider all modules as impacted.

Reply:
I didn't add it because of these comments: https://github.com/trinodb/trino/pull/10323#discussion_r800851828


From the `gib` docs:
> This property is helpful for more complex project setups in which not all submodules depend on the root module (= are not direct or indirect children of the root module).
I believe we don't have this kind of modules, right?

Reply:
No we don't have such modules as far as I can see.
However in current impl we end up running GIB when `github/workflows/ci.yml` changes for example. And GIB says that no modules changed. We'd want the reverse behaviour for safety IMO.

Reply:
It should treat this file as part of the root module and build everything.